File: go_practice/sequential_search.go
package main

import "fmt"

func main() {
	arr := []int{2, 4, 5, 10}
	var num int
	fmt.Scan(&num)
	index := sequential_search(num, arr)
	if index == -1 {
		fmt.Printf("%dは見つかりませんでした。\n", num)
	} else {
		fmt.Printf("%dは配列の%d番目のインデックスで見つかりました\n", num, index)
	}
}

func sequential_search(num int, arr []int) int {
	var index int
	for i := 0; i < len(arr); i++ {
		if arr[i] == num {
			index = i
			return index
		}
	}
	return -1
}

File: c_practice/sequential_search.c
#include <stdio.h>

int sequential_search(int, int *, int);

int main(void)
{
  int arr[] = {12, 2, 3, 43, 53};
  int len = sizeof(arr) / sizeof(arr[0]);
  int num, index;
  scanf("%d", &num);
  index = sequential_search(num, arr, len);
  if (index == -1)
  {
    printf("%dを発見できませんでした\n", num);
  }
  else
  {
    printf("%dを配列のインデックス番号%dで発見しました\n", num, index);
  }
  return 0;
}

int sequential_search(int num, int *arr, int len)
{
  for (int i = 0; i < len; i++)
  {
    if (arr[i] == num)
    {
      return i;
    }
  }
  return -1;
}
